src/fileset.rs

```diff
@@ -17,6 +17,7 @@ use crate::report::ErrorKey;
 use crate::report::{add_loaded_mod_root, error, warn_abbreviated, warn_header, will_log};
 use crate::token::{Loc, Token};
 
+/// Note that ordering of these enum values matters. Do not change the order.
```

Comment:
Instead of "do not change the order", I'd like to explain a bit more. ("Do not..." comments are a pet peeve of mine :)
Maybe "The files from later ones will override the files from earlier ones, and the error reporting is sensitive to the order when deciding what to log."

Reply:
The enum implements PartialOrd and Ord. Elsewhere in the code, enum values are compared using inequality operators. This means changing the order will fundamentally change application logic, almost definitely resulting in undesirable behaviour. This fact may not be obvious to the passer-by programmer, which is why I added the warning. There was no documentation on it previously. It is not my enum.